payload/test/test_upload.py

```diff
@@ -46,54 +47,157 @@ def setUp(self):
+                                                   chunk_count=1,
+                                                   program_name_length=10,
+                                                   program_data_length=11,
+                                                   exit_code=0,
+                                                   program_name='helloworld',
+                                                   program_output_data='helloworld\n')
+
+    def test_partially_received(self):
+        #Set up partially received program where message1 has already been received
+        shutil.copytree('uploads/testCase','uploads/programs/helloworld') 
+        message2 = self.vanguard_proto.ProgramUploadMsg.from_data(index=1, 
+                                                                  chunk=2,
+                                                                  chunk_count=3,
+                                                                  program_name_len=10,
+                                                                  program_data_len=5,
+                                                                  program_name='helloworld',
+                                                                  program_upload_data='.log(')
+
+        message3 = self.vanguard_proto.ProgramUploadMsg.from_data(index=2, 
+                                                                  chunk=3,
+                                                                  chunk_count=3,
+                                                                  program_name_len=10,
+                                                                  program_data_len=13,
+                                                                  program_name='helloworld',
+                                                                  program_upload_data='"helloworld")')
+
+        self.beacon.handle_packet(self.primary_radio,message3)
+        self.beacon.handle_packet(self.primary_radio,message2)
+
+        self.primary_radio.send.assert_called_with(type='ProgramResultMsg',
+                                                   index=0,
+                                                   chunk=1,
+                                                   chunk_count=1,
+                                                   program_name_length=10,
+                                                   program_data_length=11,
+                                                   exit_code=0,
+                                                   program_name='helloworld',
+                                                   program_output_data='helloworld\n')
+
+    def test_multi_chunk_result(self): 
+        #send a js file that sends back the contents of ./__init__.py in multiple result messages
+        with open('__init__.py', 'r') as input_file:
+            test_data = input_file.read()
+        with open('test_upload_helper.js','r') as js_file:
```

Comment:
can you use absolute paths (instead of relative) for these files? an easy way to do that is to use the builtin `__file__`, i.e:

```python
this_dir = os.path.dirname(os.path.abspath(__file__))
init_py = os.path.join(this_dir, '__init__.py')
with open(init_py, 'r') as input_file:
...
```